core/dbt/parser/manifest.py

```diff
@@ -522,11 +522,22 @@ def _check_resource_uniqueness(
         relation = relation_cls.create_from(config=config, node=node)
         full_node_name = str(relation)
 
-        existing_node = names_resources.get(name)
-        if existing_node is not None:
-            dbt.exceptions.raise_duplicate_resource_name(
-                existing_node, node
-            )
+        # Check refs - is there an unqualified ref to a
+        # model that exists in two different packages?
+        for ref in node.refs:
+            if len(ref) == 1:
+                matched = manifest.find_nodes_by_name(ref[0])
+            else:
+                # two-arg refs are namespaced, so no need
+                # to check for duplicates. Nodes must still
+                # have unique names within a package
+                continue
+
+            if len(matched) > 1:
```

Comment:
I do wonder if, in the event that there are same-named models in the current project and in an installed package, we should still support single-argument `ref` and default to the resource in the same namespace. We would only raise this error if all the resources by that name are in other namespaces.

My motivation here: It's possible that a user wants to install a package that has a same-named model as one in their project, _and_ the models in the package only uses single-argument `ref`. Of course, I think there's a lot of benefit in being more explicit to avoid surprises, so alternatively we'd establish a convention that package authors ought to use two-argument `ref` everywhere. It wouldn't be a breaking change.

Reply:
I like this idea!